src/xdocs/config_coding.xml

```diff
@@ -3160,6 +3182,40 @@ public void foo(int i, String s) {}
         <source>
 &lt;module name=&quot;PackageDeclaration&quot;/&gt;
         </source>
+        <p>
+          Examples of how the check works (matchDirectoryStructure option is false by
+          default):
+          Let us consider the AnnotationLocationCheck class which is in the directory
+          com.puppycrawl.tools.checkstyle.checks.annotations
+        </p>
+        <source>
+package com.puppycrawl.tools.checkstyle.checks;  //No Violation
+
+public class AnnotationLocationCheck extends AbstractCheck {
+//...
+}
+        </source>
+        <p>
+          Examples of how the check works if matchDirectoryStructure option is set to true:
+          Let us again consider the same class AnnotationLocationCheck.
+          The setup,
+        </p>
+        <source>
+&lt;module name=&quot;PackageDeclaration&quot;&gt;
+&lt;property name=&quot;matchDirectoryStructure&quot; value=&quot;true&quot;/&gt;
+&lt;/module&gt;
+          </source>
+          <p>
+              and the source file,
+          </p>
+          <source>
+package com.puppycrawl.tools.checkstyle.checks;
```

Comment:
please place "//violation" as trailing comment and all other text move out to `<p>`